File: orientacao-objeto/orientacao-objeto/classes/CGPessoa.m
//
//  CGPessoa.m
//  orientacao-objeto
//
//  Created by Gustavo Bergamim on 23/11/2017.
//  Copyright © 2017 Cast Group. All rights reserved.
//

#import "CGPessoa.h"

@implementation CGPessoa

@synthesize sobrenome = _sobrenome;

#pragma mark - construtor
- (instancetype) init{
    return self;
}

- (instancetype) initComNome:(NSString *)nome sobrenome:(NSString *)sobreNome
{
    self.nome = nome;
    _sobrenome = sobreNome; //synthesize
    return self;
}

#pragma mark - propertys
- (void) setNome:(NSString *)nome
{
    _nome =nome;
}


- (void) atribuirNome:(NSString *)nome sobrenome:(NSString*)param
{
    self.nome = nome;
    _sobrenome =  param; //synthesize
}

- (NSString *) nomeClasse
{
    NSLog(@"%@", NSStringFromSelector(_cmd));
    return NSStringFromClass([self class]);
}

- (void)andar {
    
}

- (void)correr {
    
}

@end

File: swift/swift-lista-pessoas/swift-lista-pessoas/CGPessoa.swift
//
//  CGPessoa.swift
//  swift-lista-pessoas
//
//  Created by Gustavo Bergamim on 12/01/18.
//  Copyright © 2018 Cast Group. All rights reserved.
//

import UIKit

class CGPessoa: NSObject {
    private (set) var nome: String;
    private (set) var idade: Int;
    private (set) var peso: Double;
    
    init(_ nome: String, _ idade: Int, _ peso: Double = 0) {
        self.nome = nome;
        self.idade = idade;
        self.peso = peso;
    }
}
